client/gutenberg/extensions/related-posts/edit.jsx

```diff
@@ -60,6 +60,70 @@ function PlaceholderPostEdit( props ) {
 	);
 }
 
+function RelatedPostsEditItem( props ) {
+	const previewClassName = 'related-posts__preview';
+
+	return (
+		<div className={ `${ previewClassName }-post` }>
+			{ props.displayThumbnails && props.post.img && props.post.img.src && (
+				<Button className={ `${ previewClassName }-post-link` } isLink>
+					<img src={ props.post.img.src } alt={ props.post.title } />
+				</Button>
+			) }
+			<h4>
+				<Button className={ `${ previewClassName }-post-link` } isLink>
+					{ props.post.title }
+				</Button>
+			</h4>
+			{ props.displayDate && (
+				<span className={ `${ previewClassName }-post-date has-small-font-size` }>
+					{ props.post.date }
+				</span>
+			) }
+			{ props.displayContext && (
+				<p className={ `${ previewClassName }-post-context` }>{ props.post.context }</p>
+			) }
+		</div>
+	);
+}
+
+function RelatedPostsPreviewRows( props ) {
+	const className = 'related-posts__row';
+
+	let topRowEnd = [];
+	let bottomRowStart = [];
+	const displayLowerRow = props.posts.length > 3;
+
+	switch ( props.posts.length ) {
+		case 2:
+		case 4:
+		case 5:
+			topRowEnd = 2;
+			break;
+		default:
+			topRowEnd = 3;
+			break;
+	}
+	switch ( props.posts.length ) {
```

Comment:
I think we could combine the 2 switch statements into one here.

Reply:
I think it is more readable this way.

Reply:
Hmm, that's a lot of `switch`aroo :sweat_smile: 

AFAICS, currently, `bottomRowStart === topRowEnd`. Isn't that always going to be the case? If it is, can we simply remove one of the variables and use the other instead?

Reply:
> AFAICS, currently, bottomRowStart === topRowEnd. Isn't that always going to be the case? If it is, can we simply remove one of the variables and use the other instead?

Now I feel very silly.

Reply:
@ockham and @enejb  — You are a gentlemen and scholars and I feel extremely privileged for working with you.